packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraSettingsTests.m

```diff
@@ -0,0 +1,244 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera_avfoundation;
+@import camera_avfoundation.Test;
+@import XCTest;
+@import AVFoundation;
+#import <OCMock/OCMock.h>
+#import "CameraTestUtils.h"
+#import "MockFLTThreadSafeFlutterResult.h"
+
+static const char *gTestResolutionPreset = "medium";
+static const int gTestFPS = 15;
+static const int gTestVideoBitrate = 200000;
+static const int gTestAudioBitrate = 32000;
+static const bool gTestEnableAudio = YES;
+
+@interface PositiveNumberOnNilTests : XCTestCase
+@end
+
+/// Expect that optional positive numbers can be parsed
+@implementation PositiveNumberOnNilTests
+
+- (void)testPositiveOrNilShouldRejectNegativeIntNumber {
+  NSError *error;
+  NSNumber *parsed = [CameraPlugin positiveOrNil:@(-100) error:&error];
+  XCTAssert(parsed == nil && error, "should reject negative int number");
+}
+
+- (void)testPositiveOrNilShouldRejectNegativeFloatingPointNumber {
+  NSError *error;
+  NSNumber *parsed = [CameraPlugin positiveOrNil:@(-3.7) error:&error];
+  XCTAssert(parsed == nil && error, "should accept positive floating point number");
+}
+
+- (void)testNanShouldBeParsedAsNil {
+  NSError *error;
+  NSNumber *parsed = [CameraPlugin positiveOrNil:@(NAN) error:&error];
+  XCTAssert(parsed == nil && error, "should reject NAN");
+}
+
+- (void)testPositiveOrNilShouldRejectNil {
+  NSError *error;
+  NSNumber *parsed = [CameraPlugin positiveOrNil:nil error:&error];
+  XCTAssert(parsed == nil && !error, "should accept nil");
+}
+
+- (void)testPositiveOrNilShouldAcceptNull {
+  NSError *error;
+  NSNumber *parsed = [CameraPlugin positiveOrNil:[NSNull null] error:&error];
+  XCTAssert(parsed == nil && !error, "should accept [NSNull null]");
+}
+
+- (void)testPositiveOrNilShouldAcceptPositiveFloatingPointNumber {
+  NSError *error;
+  NSNumber *parsed = [CameraPlugin positiveOrNil:@(3.7) error:&error];
+  XCTAssert(parsed != nil && !error, "should accept positive floating point number");
+}
+
+- (void)testPositiveOrNilShouldAcceptPositiveDecimalNumber {
+  NSError *error;
+  NSNumber *parsed = [CameraPlugin positiveOrNil:@(5) error:&error];
+  XCTAssert(parsed != nil && !error, "should parse positive int number");
+}
+
+@end
+
+@interface CameraSettingsTests : XCTestCase
+@end
+
+@interface TestMediaSettingsProvider : FLTCamMediaSettingsProvider
```

Comment:
This needs a documentation comment explaining what it does and how it fits into the test structure.